python/lsst/daf/butler/core/_butlerUri/_butlerUri.py

```diff
@@ -621,6 +621,17 @@ def isabs(self) -> bool:
         """
         return True
 
+    def abspath(self) -> ButlerURI:
+        """Return URI using an absolute path.
+
+        Returns
+        -------
+        abs : `ButlerURI`
+            Absolute URI. For non-schemeless URIs this always returns itself.
+            Schemeless URIs are upgraded to file URIs.
+        """
```

Comment:
Is there anything you could `assert` on here to make sure a new derived class that needs to implement this method doesn't forget to?

Reply:
All URIs are by definition meant to be absolute so this should always be the right answer. It's only schemeless that needs to be upgraded to absolute `file` URI (`file` is always absolute).